### PR TITLE
chore(governance): add in some basic things for maintenance

### PR DESCRIPTION
**Pull request by ckipp01**
This is a PR that came about during the BSP discussions at the Scala
Tooling Summit. This change updates the README and the CONTRIBUTING doc
with some more relevant information about how this project is
maintained and also added some guidelines. It also adds a CODEOWNERS and
a feature request template.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (ckipp01) [inline .github/CODEOWNERS:1]: So the idea here is that whenever someone updates the actual protocol (anything touching the xtend files) we'll all be automatically assigned to review. We can either review ourselves or add someone else from out teams to do this.

File: readme.md
Comment (ckipp01) [inline readme.md:26]: I added the two of you here since you've both contributed somewhat frequently to the repo lately. Is this alright? Also @abrams27, I know you don't have a ton of commits to the repo, but also know you're heavily involved in the BSP/Jetbrains side of things. Do you also want to be listed here, or would you rather us leave you off?

File: .github/CODEOWNERS
Comment (adpi2) [reply .github/CODEOWNERS:1]: That's a nice feature.

File: CONTRIBUTING.md
Comment (adpi2) [inline CONTRIBUTING.md:13]: I would turn this sentence into a recommendation instead of a requirement.

For some small changes, I feel like it can be a waste of time to do it in two steps: open an issue, wait for approval, then work on the PR, wait for a second approval. A PR with a good description can save time of both the contributor and the reviewer, even if there is a chance the PR get rejected.

**Review by adpi2 — COMMENTED** [review]
That's great!

As a side comment, I wonder how we can make sure that all PRs are reviewed in due time. What should we do if some PR is waiting for approval for more than 2 weeks or a month?

File: CONTRIBUTING.md
Comment (ckipp01) [reply CONTRIBUTING.md:13]: Ah, so I more meant when you add an entire new feature like `buildTarget/formatting` or something that you do need a discussion. I figured small improvements, etc implicitly don't need one. I'll make this clearer in there.

**Comment by ckipp01** [comment]
> As a side comment, I wonder how we can make sure that all PRs are reviewed in due time. What should we do if some PR is waiting for approval for more than 2 weeks or a month?

Good point. I'm not really sure. We can just like put a note in there or something saying that within 2 weeks if someone from that represented party doesn't respond it's assumed they are ok with it?

**Review by adpi2 — APPROVED** [review]

File: .github/CODEOWNERS
Comment (jastice) [reply .github/CODEOWNERS:1]: shall we add smithy files in here right now to not forget it later?

File: CONTRIBUTING.md
Comment (jastice) [inline CONTRIBUTING.md:14]: I feel it should be okay to submit a draft PR as a basis of discussion, with the expectation that it might not be accepted or heavily reworked before merging. Sometimes it's easier to just write / look at the code rather than laying out an abstract plan in detail.

File: readme.md
Comment (jastice) [inline readme.md:7]: JetBrains

File: .github/CODEOWNERS
Comment (ckipp01) [reply .github/CODEOWNERS:1]: There might be some small edits as we merge smithy in and tweak things. So I'd recommend getting it in first, making sure it's working as expected, and then we can rid the codebase of any and all `xtend` references and replace them with smithy.

File: CONTRIBUTING.md
Comment (ckipp01) [reply CONTRIBUTING.md:14]: Good point. I went ahead and re-worked this section.

File: .github/ISSUE_TEMPLATE/feature_request.yml
Comment (agluszak) [inline .github/ISSUE_TEMPLATE/feature_request.yml:38]: Doesn't marking it as required turn off the possibility to leave it blank?

File: readme.md
Comment (agluszak) [inline readme.md:11]:
```suggestion
Even though the protocol currently focuses on Scala developer tools, it's designed to
```

File: .github/ISSUE_TEMPLATE/feature_request.yml
Comment (ckipp01) [reply .github/ISSUE_TEMPLATE/feature_request.yml:38]: Ah duh. Yea you're right. I'll remove these.

**Review by jastice — APPROVED** [review]

File: readme.md
Comment (abrams27) [reply readme.md:26]: sorry for the delay (last week was really busy), u can add me later if it works for u

Comment (ckipp01) [reply readme.md:26]: No worries at all! Feel free to PR yourself in.

**Pull request merged**
